yt/yt/server/queue_agent/consumer_controller.cpp

```diff
@@ -388,9 +403,8 @@ class TConsumerSnapshotBuildSession final
 
             const auto& partitionRowInfo = partitionInfo.begin()->second;
 
-            if (partitionRowInfo.CumulativeDataWeight) {
+            if (!consumerPartitionSnapshot->CumulativeDataWeight && partitionRowInfo.CumulativeDataWeight) {
```

Comment:
We do not really need to check the second condition here, assigning the optional altogether is fine.